tests/test_config.py

```diff
@@ -703,6 +703,33 @@ def test_config_precedence():
     assert cfg(f=LOCAL)['pytest']['precedence'] == 'local'
 
 
+def test_config_non_readable_file(config_tmpdir):
+    # test to read a config file without read permission
+    # Note: non-readable files do not work on Windows
+    if WINDOWS:
```

Comment:
Off-topic sorry but I just found in https://docs.pytest.org/en/6.2.x/skipping.html#skipping-test-functions that `pytest` uses a different, "fuzzier" test to determine if the OS is Windows... worth looking into? In a different PR.